### PR TITLE
Add credential refresh logic for long running tests

### PR DESCRIPTION
**Pull request by phani-srikar**
#### Description of changes

- Add refresh credentials for some Searchable tests that could take more than an hour to run. This eliminates the token expiry due to role chaining.
- Use `amplify-integ-test-user` profile to collect flaky test metrics. This step used to fail if job runs for more than an hour.
- Updates to codebuild split-e2e script to use parent account for searchable tests.

#### Description of how you validated changes
Verified that `searchable-migration` test runs beyond an hour and passes.

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [ ] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#tests)
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] New AWS SDK calls or CloudFormation actions have been added to relevant test and service IAM policies
- [ ] Any CDK or CloudFormation parameter changes are called out explicitly

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-e2e-tests/refresh-credentials.sh [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+creds=$(aws sts assume-role --role-arn $TEST_ACCOUNT_ROLE --role-session-name testSession1248634 --duration-seconds 3600)
+if [ -z $(echo $creds | jq -c -r '.AssumedRoleUser.Arn') ]; then
+    return
+fi
+
+export AWS_ACCESS_KEY_ID=$(echo $creds | jq -c -r ".Credentials.AccessKeyId")
+export AWS_SECRET_ACCESS_KEY=$(echo $creds | jq -c -r ".Credentials.SecretAccessKey")
+export AWS_SESSION_TOKEN=$(echo $creds | jq -c -r ".Credentials.SessionToken")
+
+echo "{ \"accessKeyId\": \"$AWS_ACCESS_KEY_ID\", \"secretAccessKey\": \"$AWS_SECRET_ACCESS_KEY\", \"sessionToken\": \"$AWS_SESSION_TOKEN\" }"
```
Comment (AaronZyLee) [inline packages/amplify-e2e-tests/refresh-credentials.sh:21]: Is the echo here used for the shell return value? Will this echo get exposed in the console?

File: packages/amplify-e2e-core/package.json [hunk 2]
```diff
@@ -39,7 +39,9 @@
     "semver": "^7.3.5",
     "strip-ansi": "^6.0.0",
     "throat": "^5.0.0",
-    "uuid": "^8.3.2"
+    "uuid": "^8.3.2",
+    "ini": "^3.0.1",
+    "@types/ini": "^1.3.31"
```
Comment (alharris-at) [inline packages/amplify-e2e-core/package.json:44]: should be a dev dependency?

File: packages/amplify-e2e-tests/src/__tests__/graphql-v2/searchable-datastore.test.ts [hunk 3]
```diff
@@ -27,7 +31,8 @@ describe('transformer model searchable migration test', () => {
   });
 
   afterEach(async () => {
-    await deleteProject(projRoot);
+    const newCreds = refreshCredentials();
```
Comment (AaronZyLee) [inline packages/amplify-e2e-tests/src/__tests__/graphql-v2/searchable-datastore.test.ts:34]: Can we add a codebuild ci check here? The credential refresh is only needed in CodeBuild CI and not needed if running locally or in CircleCI. The current env variables used are `process.env.CI && process.env.CODEBUILD`

**Review by AaronZyLee — CHANGES_REQUESTED** [review]

File: packages/amplify-e2e-tests/src/__tests__/graphql-v2/searchable-datastore.test.ts (hunk 3 above)
Comment (phani-srikar) [reply packages/amplify-e2e-tests/src/__tests__/graphql-v2/searchable-datastore.test.ts:34]: Added an early return inside the refresh method, that should take care of this.

File: packages/amplify-e2e-core/package.json (hunk 2 above)
Comment (phani-srikar) [reply packages/amplify-e2e-core/package.json:46]: moved the types to dev-dep. Thanks.

File: packages/amplify-e2e-tests/refresh-credentials.sh (hunk 1 above)
Comment (phani-srikar) [reply packages/amplify-e2e-tests/refresh-credentials.sh:21]: Yes. Not unless we run this script outside of a child process. Hence I put this inside the `e2e-tests` package different from the other `shared-scripts.sh` that we run directly from the shell.

**Review by AaronZyLee — APPROVED** [review]

**Review by alharris-at — APPROVED** [review]

File: packages/amplify-e2e-tests/refresh-credentials.sh (hunk 1 above)
Comment (sobolk) [inline packages/amplify-e2e-tests/refresh-credentials.sh:21]: why not https://docs.aws.amazon.com/sdk-for-javascript/v3/developer-guide/javascript_sts_code_examples.html ?

**Pull request merged**